### DISCUSSION TITLE
Using format expression for "uses" value?

### DISCUSSION THREAD

**Discussion by mandrean** [community/community#25446] [Actions]
<p>In my workflow file, I have a step like below:</p>
<pre><code class="lang-auto">      - name: Semantic Release
        uses: ${{ format('docker://{0}', github.repository) }}
        with:
          args: -r yes -w yes
</code></pre>
<p>Basically I’m trying to format a string into <code>docker://&lt;image-name&gt;</code> based on what the GitHub repo’s owner/name is.</p>
<p>But I get this error:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/8/38b219d2af6f5cc9a62018e0071dbec1947b4172.png" data-download-href="/uploads/short-url/85yjHfaSiVgDTulRZgUeuYtr1Lk.png?dl=1" title="Skärmavbild 2020-10-16 kl. 15.57.54" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/4631744/181091164-ae9b9b1b-8f18-4c18-93a2-3179cb8fd3f1.png" alt="Skärmavbild 2020-10-16 kl. 15.57.54" data-base62-sha1="85yjHfaSiVgDTulRZgUeuYtr1Lk" width="690" height="147" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38b219d2af6f5cc9a62018e0071dbec1947b4172_2_690x147.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38b219d2af6f5cc9a62018e0071dbec1947b4172_2_1035x220.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38b219d2af6f5cc9a62018e0071dbec1947b4172_2_1380x294.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38b219d2af6f5cc9a62018e0071dbec1947b4172_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Skärmavbild 2020-10-16 kl. 15.57.54</span><span class="informations">2038×436 53.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<blockquote>
<p><code>The workflow is not valid. .github/workflows/ci.yml (Line: 72, Col: 15): Unrecognized named-value: 'github'. Located at position 24 within expression: format('docker://{0}', github.repository)</code></p>
</blockquote>
<p>Is it even possible to do what I’m trying to do? I’ve also tried just:</p>
<pre><code class="lang-auto">      - name: Semantic Release
        uses: docker://${{ github.repository) }}
        with:
          args: -r yes -w yes
</code></pre>
<p>But same result.</p>

**Comment by Simran-B** [comment, marked as answer]
<p>The <code>uses</code> value cannot be set dynamically:</p>
<aside class="quote" data-post="2" data-topic="17466">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/env-variables-in-uses/17466/2">Env variables in uses</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    In order to enact policies like only using actions defined in the org or repo we can’t allow actions to dynamically change at runtime.  So using any sort of dynamic value in uses is not something we will be able to support.
  </blockquote>
</aside>

**Comment by mandrean** [comment]
<p>I see! Thanks for the reply.</p>
